### ISSUE TITLE
[UI/UX] Rescan only a single rom / Rescan Size/Reg/Lang/Rev without a full rescan

### ISSUE THREAD

**Issue by VFansss** [rommapp/romm#1018]
Hi! It's not the place where to discuss this, but I would thank you for all your work, Romm is amazing: it's almost exactly what I wanted to realize by myself years ago but I hadn't the willpower (and knowledge) to do.

I'm using the latest 3.3.0 release, and I'm trying to "match" my games into the Romm database.

The scan functionality works good, and with the new version you can also fine-tune the name to search something that is wrongly recognized or slightly offset, so kudos +++ for that!

However, I'm experimenting with rom naming convention (I want to explicit both Region and Lang on the rom name, like the example below) and would honestly much better to have the flexibility to further rescan only the rom I'm actually viewing instead of everything.

Because you can't edit these property manually (and in my opinion is a correct choice) so you are forced to change the rom name to do so (and that's correct too, in my opinion) however seems that changing the rom name doesn't refresh basically anything of the game, so you have to force a rescan somehow.

![immagine](https://github.com/user-attachments/assets/f69e4507-d491-42b8-9aac-a2fcda74b2f8)

However, using the "rescan" functionality (/scan) to refresh the rom works ONLY selecting the "complete rescan" option: every other option doesn't do anything about rom with changed names.

The whole rescan is quite slow, could break other roms parsing and in general isn't a good idea to refresh the whole platform just for a single changed rom.

Is possible to have some button to force a rescan on a single rom, to update Size/Reg/Lang/Rev?

As a side question: how Reg/Lang/Rev are parsed by Romm? I can't find any official guide, and trial-and-error doesn't seems to be very effective (the reason is explained above!)

Thanks for all your works, for me it's a dream software, I would like to contribute somehow when I will have free time...

**Comment by VFansss** [comment]
Thinking about this, we could "rewrite" this issue in "refresh Name, Size/Reg/Lang/Rev" when editing the rom property from the rom page, instead of relying of doing a scan/full rescan.

**Comment by gantoine** [comment]
From the gallery view, you can select multiple games and only rescan those games. Looking at the code, even a "quick scan" should pickup the change in tags/lang/rev and filesize.

<img width="427" alt="Screenshot 2024-07-24 at 11 57 21 AM" src="https://github.com/user-attachments/assets/eff1cf98-f7b8-4a66-8393-6d2e615ad3c1">

**Comment by VFansss** [comment]
Ehi there! Thanks for the quick answer!

You are right, I didn't thought about the list multi-selection (that's why I suspected was a good idea to mark this issue as UI/UX instead of a feature: I guess everything that's there, it just have to be accessed the best way).

I've tried it and I have to say that's working good...almost. Because actually it rescan more roms than the selected one.

In fact, if I try to change the filename from

`Zelda II - The Adventure of Link (USA) [IT][Traduzione 1.0].nes`

to 

`Zelda II - The Adventure of Link (USA) [FR][Traduzione 1.0].nes`

I select it with multiselect and then I start rescanning, it actually rescan...more games (weirdly, in the list there's not the game I've told to rescan, and in fact it take a while before finish. However, refreshing the page at the end actually refresh the game tags on the game I've selected to rescan, so we can say it actually works somehow)

![immagine](https://github.com/user-attachments/assets/f4e4f1d2-a25f-47f3-a204-929fbed917b2)

![immagine](https://github.com/user-attachments/assets/a6277887-eca2-42df-ac80-2b845b0b6b24)

![immagine](https://github.com/user-attachments/assets/8060de71-afac-451d-b7fb-0f2678d82d87)

> Looking at the code, even a "quick scan" should pickup the change in tags/lang/rev and filesize.

In my case, as previously said, it don't pickup changes with any option than "full rescan" (either on every platform on only the actually changed one)

In the end, I'm happy that I can quickly mess with tags, and even rename roms directly from the UI.

I will leave to you the decision if worth investigating further or close the issue.

In both cases, thanks for your efforts, I'm loving Romm <3
